### PR TITLE
:seedling: Fix bad smells

### PR DESCRIPTION
**Pull request by cupakob**
## Related issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by embik** [comment]
Hi @cupakob, thank you for the contribution! Before we can accept your contribution, please make sure it is signed off (the `-s` flag on `git commit`) so you signal that you are accepting the [DCO](https://github.com/kcp-dev/kcp/blob/main/DCO). In addition, it would be great if you could write a short summary (how are the things you've changed smells?) so that this PR has the necessary context.

**Comment by cupakob** [comment]
> how are the things you've changed smells?

done

File: test/e2e/fixtures/webhook/webhook.go [hunk 1]
```diff
@@ -59,7 +60,7 @@ func (s *AdmissionWebhookServer) StartTLS(t *testing.T, certFile, keyFile string
 
 	go func() {
 		err := serv.ListenAndServeTLS(certFile, keyFile)
-		if err != nil && err != http.ErrServerClosed {
+		if err != nil && errors.Is(err, http.ErrServerClosed) {
```
Comment (embik) [inline test/e2e/fixtures/webhook/webhook.go:63]: This looks like a change in logic to me (from "err is **not** `http.ErrServerClosed`" to "err _is_ `http.ErrServerClosed`"). Is that intentional?

**Review by embik — CHANGES_REQUESTED** [review]

File: test/e2e/fixtures/webhook/webhook.go (hunk 1 above)
Comment (cupakob) [reply test/e2e/fixtures/webhook/webhook.go]: no, its fixed now

**Review by embik — APPROVED** [review]
/approve

Thank you for contributing to kcp!

**Pull request merged**